test/livebook_web/live/home_live_test.exs

```diff
@@ -172,6 +172,27 @@ defmodule LivebookWeb.HomeLiveTest do
     assert render(view) =~ "Welcome to Livebook"
   end
 
+  describe "notebook import" do
+    test "allows importing notebook directly from content", %{conn: conn} do
+      Phoenix.PubSub.subscribe(Livebook.PubSub, "sessions")
+
+      {:ok, view, _} = live(conn, "/home/import/content")
+
+      notebook_content = """
+      # My notebook
+      """
+
+      view
+      |> element("form", "Import")
+      |> render_submit(%{data: %{content: notebook_content}})
+
+      assert_receive {:session_created, id}
+
+      {:ok, view, _} = live(conn, "/sessions/#{id}")
+      assert render(view) =~ "My notebook"
+    end
```

Comment:
Should we add a test about what happens when the notebook is invalid?

Reply:
When importing notebook we always consider the content valid. We do best effort when parsing live markdown and return a list of warnings.

Reply:
So let’s add a test about this case. But I wonder what would happen if it is fully invalid, like an HTML content.

Reply:
For a simple HTML like this:

```html
<!DOCTYPE html>
<html lang="en">
<head>
  <title>Document</title>
</head>
<body>
  <h1>Page title</h1>
  <div>
    <p>Some content</p>
    <p>More content</p>
  </div>
  <button>Button text</button>
</body>
</html>
```

We end up with a single markdown cell:

![image](https://user-images.githubusercontent.com/17034772/115750748-cd522000-a398-11eb-811e-481d5d734f93.png)

In this case there are no warnings because it's actually just an HTML tag and that's valid extended markdown.

Generally arbitrary text *could* be Markdown, so validating the correctness is probably tricky anyway.

---

When loading from URL we check for the content type and if it's HTML we will show an error. For the paste import, if the user explicitly puts HTML there, then they make an explicit choice of not getting a reasonable notebook out of that.

Reply:
I wonder if we should make our importer a bit more strict and require the first non-empty line to be a title. This can be done by traversing until we find a #. Just to avoid cases like these that are clearly not markdown?

Reply:
It kinda depends on what we want to achieve. Currently the import is doing its best to also parse non-livemarkdown markdown reasonably. For example the user may import a regular markdown file without `#` in which case we just assume the default notebook name but properly parse the remaining content.

Reply:
I can imagine such notebook:

---

# HTML parser in Elixir

This notebook explores writing a parser that converts this:

```html
<html lang="en">
  <!-- ... -->
</html>
```

Into this:

```elixir
[{:html, [lang: "en"], []}]
```

Reply:
> So what do you think about looking for <html explicitly and whatever people use to start HTML. That’s the most likely mistake to happen, so we should at least protect against those.

But as said, when loading from URL we check the content type, so it would need to be a URL pointing to raw HTML (or the user would paste HTML explicitly). I think the content type check should cover the common mistake path, right?

Reply:
Ah, good call. Ok, sounds good to me!

Reply:
To be clear, I would check only the first characters for `<html` not the whole document - but I think the content type will suffice for now. :)